TWUITests/APIStubs/HTTPDynamicStubbs.swift

```diff
@@ -58,23 +61,19 @@ final class HTTPDynamicStubs: HTTPDynamicStubing {
         setupStub(stubInfo)
     }
 
-    func replaceValues(of items: [String: String], in stub: APIStubInfo) {
+    func replace(with modification: RegexJSONModifier.Modification, in stub: APIStubInfo) {
+        transform({
+            self.regexModifier.apply(modification: modification, in: $0)
+        }, in: stub)
+    }
+
+    private func transform(_ modifyFn: (Data) -> Data?, in stub: APIStubInfo) {
         do {
             let dataObject = getDataObject(from: stub)
             guard let json = dataToJSON(data: dataObject) else { return }
             let data = try JSONSerialization.data(withJSONObject: json, options: JSONSerialization.WritingOptions.prettyPrinted)
-            guard var string = String(data: data, encoding: .utf8) else { return }
-
-            for (key, value) in items {
-                string = string.replacingOccurrences(
-                    of: "\"\(key)\"\\s?:\\s?\".*\"",
-                    with: "\"\(key)\" : \"\(value)\"",
-                    options: .regularExpression)
-            }
-
-            if let data = string.data(using: .utf8) {
-                stubJSON(object: data, for: stub)
-            }
+            guard let modifiedData = modifyFn(data) else { return }
```

Comment:
maybe `try modifyFn` and throw error. This error can then be catched same way as `JSONSerialization.data(...`